### DISCUSSION TITLE
Importing rtl_433 device decoders

### DISCUSSION THREAD

**Discussion by dwmw2** [dbuezas/esphome-cc1101#30] [Ideas]
There are a bunch of device decoders in the rtl_433 project, including one or two I reverse-engineered and contributed myself like https://github.com/merbanan/rtl_433/blob/master/src/devices/oil_watchman.c 

I've actually been running rtl_433 itself for years on my OpenWrt router, chewing its CPU doing SDR and vaguely planning to replace it with an RFM69 or CC1101. Your esphome-cc1101 is the ideal way to do so. But... wouldn't it be nice if we could just use the rtl_433 *device* decoders, a bit like `remote_receiver` protocols?

They're fairly self-contained, and mostly all we need to do is set up the CC1101 appropriately then feed them a `bitbuffer` of data in the form they expect. And then we'd magically have support for all those devices in ESPHome?

The device abstraction in rtl_433 is fairly stable; it shouldn't be hard to get them to expose it as a library (even just make it possible to *build* it as a static library, without making any ABI promises, so we could import it from a custom component).

Am I being too naïve?

**Comment by dwmw2** [comment]
I asked here about exposing the device decoders as some kind of library: https://github.com/merbanan/rtl_433/discussions/3073

**Reply by dbuezas** [reply]
I think rtl_433 can decode other more complex modulations too, not just OOK (one-off keying)

**Reply by dwmw2** [reply]
Yes, it can definitely do FSK. I'm going to play with https://github.com/NorthernMan54/rtl_433_ESP which was mentioned in the other discussion (and on Discord).
